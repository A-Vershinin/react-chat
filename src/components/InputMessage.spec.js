/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import InputMessage from './InputMessage.jsx';

const mockProps = {
  classes: {
    messageInputWrapper: 'some text',
    messageInput: 'some text 2',
  },
  onJoinButtonClick: jest.fn(),
  showJoinButton: false,
  disabled: false,
};

describe('<InputMessage />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <InputMessage {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <InputMessage {...mockProps} />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders when chatItem disabled', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <InputMessage {...mockProps} disabled />
              </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders when show joinButton', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <InputMessage {...mockProps} showJoinButton />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
