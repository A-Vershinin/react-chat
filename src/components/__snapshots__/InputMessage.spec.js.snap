// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputMessage /> renders correctly 1`] = `
<div
  className="InputMessage-messageInputWrapper-43 some text"
>
  <div
    className="MuiPaper-root-45 MuiPaper-elevation6-53 MuiPaper-rounded-46 InputMessage-messageInput-44 some text 2"
  >
    <div
      className="MuiInput-root-72 MuiInput-fullWidth-79 MuiInput-underline-76"
      onKeyPress={[Function]}
    >
      <input
        aria-invalid={undefined}
        aria-required={undefined}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-80"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Type your message…"
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<InputMessage /> renders when chatItem disabled 1`] = `
<div
  className="InputMessage-messageInputWrapper-43 some text"
>
  <div
    className="MuiPaper-root-45 MuiPaper-elevation6-53 MuiPaper-rounded-46 InputMessage-messageInput-44 some text 2"
  >
    <div
      className="MuiInput-root-72 MuiInput-disabled-75 MuiInput-fullWidth-79 MuiInput-underline-76"
      onKeyPress={[Function]}
    >
      <input
        aria-invalid={undefined}
        aria-required={undefined}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-80 MuiInput-disabled-75"
        defaultValue={undefined}
        disabled={true}
        id={undefined}
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Type your message…"
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<InputMessage /> renders when show joinButton 1`] = `
<div
  className="InputMessage-messageInputWrapper-43 some text"
>
  <div
    className="MuiPaper-root-45 MuiPaper-elevation6-53 MuiPaper-rounded-46 InputMessage-messageInput-44 some text 2"
  >
    <button
      className="MuiButtonBase-root-107 MuiButton-root-85 MuiButton-contained-94 MuiButton-containedPrimary-95 MuiButton-raised-97 MuiButton-raisedPrimary-98 MuiButton-fullWidth-106"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-86"
      >
        Join
      </span>
      <span
        className="MuiTouchRipple-root-110"
      />
    </button>
  </div>
</div>
`;
